tools/tasks.py

```diff
@@ -24,6 +24,19 @@
 import urllib.request
 
 
+# The 'latin-1' encoding to map byte values directly to the first 256 Unicode code points.
+# This is the closest equivalent Python 3 offers to the permissive Python 2 text handling model.
+# While the Windows cp1252 encoding is also sometimes referred to as 'latin-1', it doesn't map
+# all possible byte values, and thus needs to be used in combination with the 'surrogateescape'
+# error handler to ensure it never throws UnicodeDecodeError.
+ENCODING_LATIN1 = 'latin-1'
+
+# This is the error handler that Python uses for most OS facing APIs to gracefully cope with encoding
+# problems in the data supplied by the OS. It handles decoding errors by squirreling the data away in
+# a little used part of the Unicode code point space. When encoding, it translates those hidden away
+# values back into the exact original byte sequence that failed to decode correctly.
```

Comment:
Again, this is verbatim the python docs, and is describing what the option does, we can always look it up if nessesary. Indicating _why_ we're including this specific handler is the more relevant information for us to include in a comment.

I actually feel like we should be using `backslashreplace` instead, to escape invalid byte sequences, rather than replace them with something else.

Reply:
Used backslashreplace to escape byte sequences, rather than replace them with something else.